modules/sfm/CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+
+if (NOT HAVE_EIGEN)
+  message(STATUS "Module opencv_sfm disabled because Eigen library is required.")
+  ocv_module_disable(sfm)
+  return()
+endif()
+
+set(the_description "SFM algorithms")
+
+ocv_warnings_disable(CMAKE_CXX_FLAGS -Wundef -Winvalid-pch)
+
+if(UNIX)
+  if(CMAKE_COMPILER_IS_GNUCXX OR CV_ICC)
+     set(CMAKE_CXX_FLAGS "${CMAKE_CSS_FLAGS} -fPIC")
+  endif()
+endif()
+
+
+### LIBMV LIGHT DEFINITIONS ###
+
+set(LIBMV_LIGHT_INCLUDES
+  src/libmv_light
+  src/libmv_light/third_party/ssba
+  "${OpenCV_SOURCE_DIR}/include/opencv"
+)
+
+set(LIBMV_LIGHT_LIBS
+  reconstruction
+  correspondence
+  camera
+  multiview
+  multiview_test_data
+  numeric
+)
+
+
+### LIBMV LIGHT EXTERNAL DEPENDENCIES ###
+
+find_package( Ceres QUIET )
+
+if ( Ceres_FOUND )
+  add_definitions("-DCERES_FOUND=1")
+  list(APPEND LIBMV_LIGHT_LIBS simple_pipeline)
+else ()
+  add_definitions("-DCERES_FOUND=0")
+  message(STATUS "CERES support is disabled. Ceres Solver for reconstruction API is required.")
+endif ()
+
+
+### DEFINE OPENCV SFM MODULE DEPENDENCIES ###
+
+set(OPENCV_SFM_DEPS
+  opencv_core
+  opencv_calib3d
+  opencv_features2d
+  opencv_xfeatures2d
+)
+
+
+### CREATE OPENCV SFM MODULE ###
+
+ocv_add_module(sfm ${OPENCV_SFM_DEPS})
+ocv_include_directories( ${LIBMV_LIGHT_INCLUDES} )
+
+# build libmv_light
+add_subdirectory(src/libmv_light)
+
+# source files
+FILE(GLOB OPENCV_SFM_SRC src/*.cpp)
+
+# define the header files (make the headers appear in IDEs.)
+FILE(GLOB OPENCV_SFM_HDRS include/opencv2/sfm/*.hpp)
+
+ocv_set_module_sources( HEADERS ${OPENCV_SFM_HDRS} SOURCES ${OPENCV_SFM_SRC} )
+
+ocv_module_include_directories()
```

Comment:
Place `ocv_module_include_directories()` on line 65 (before `add_subdirectory(src/libmv_light)`)
This should resolve this problem:
```
opencv_contrib/modules/sfm/src/libmv_light/libmv/correspondence/feature.h:24:28: fatal error: opencv2/core.hpp: No such file or directory
 #include <opencv2/core.hpp>
```